packages/cubejs-databricks-jdbc-driver/src/DatabricksDriver.ts

```diff
@@ -262,38 +271,46 @@ export class DatabricksDriver extends JDBCDriver {
 
     super(config);
     this.config = config;
+    this.parsedConnectionProperties = parseDatabricksJdbcUrl(url);
     this.showSparkProtocolWarn = showSparkProtocolWarn;
   }
 
-  /**
-   * @override
-   */
-  public readOnly() {
+  public override readOnly() {
     return !!this.config.readOnly;
   }
 
-  /**
-   * @override
-   */
-  public capabilities(): DriverCapabilities {
+  public override capabilities(): DriverCapabilities {
     return {
       unloadWithoutTempTable: true,
       incrementalSchemaLoading: true
     };
   }
 
-  /**
-   * @override
-   */
-  public setLogger(logger: any) {
+  public override setLogger(logger: any) {
     super.setLogger(logger);
     this.showDeprecations();
   }
 
-  /**
-   * @override
-   */
-  public async loadPreAggregationIntoTable(
+  public override async testConnection() {
+    const token = `Bearer ${this.config.properties.PWD}`;
+
+    const res = await fetch(`https://${this.parsedConnectionProperties.host}/api/2.0/sql/warehouses/${this.parsedConnectionProperties.warehouseId}`, {
+      headers: { Authorization: token },
+    });
+
+    if (!res.ok) {
+      throw new Error(`Databricks API error: ${res.statusText}`);
+    }
+
+    // const _data = await res.json();
+
+    // We don't need to check the actual status, because SQL WH will auto resume on real connection
+    // if (data.state !== 'RUNNING') {
```

Comment:
I think it should check anyway, at least for non `DELETING | DELETED` status

Also `health` field looks interesting, like when `health.status` is `FAILED` we can fail early this test.
https://docs.databricks.com/api/workspace/warehouses/get#health

Reply:
All right! You got it!) Let's make it good! :)